### ISSUE TITLE
Scan completed but no result displayed

### ISSUE THREAD

**Issue by syafsss** [Secure-Compliance-Solutions-LLC/GVM-Docker#26]
Hi, 

We have downloaded the latest docker image with this [commit](https://github.com/Secure-Compliance-Solutions-LLC/GVM-Docker/commit/39bf6c2de79e5f7a47c93e155fafe8e2a65b0586) and performed a scan on our host target. 

While waiting for GVM SecInfo to fully load, we came across below error, but was still able to perform scan, so we proceed scanning the target until its completed: 

> md manage:WARNING:2020-05-22 06h40.13 utc:4489: sql_open: PQerrorMessage (conn): FATAL:  could not open relation mapping file "global/pg_filenode.map": Too many open files in system
md manage:WARNING:2020-05-22 06h40.13 utc:4489: init_manage_process: sql_open failed
md manage:WARNING:2020-05-22 06h40.13 utc:4492: sql_open: PQconnectPoll failed
md manage:WARNING:2020-05-22 06h40.13 utc:4492: sql_open: PQerrorMessage (conn): FATAL:  could not open relation mapping file "global/pg_filenode.map": Too many open files in system
md manage:WARNING:2020-05-22 06h40.13 utc:4492: init_manage_process: sql_open failed
md manage:WARNING:2020-05-22 06h40.16 utc:4494: sql_open: PQconnectPoll failed
md manage:WARNING:2020-05-22 06h40.16 utc:4494: sql_open: PQerrorMessage (conn): FATAL:  could not open relation mapping file "global/pg_filenode.map": Too many open files in system
md manage:WARNING:2020-05-22 06h40.16 utc:4494: init_manage_process: sql_open failed
md manage:WARNING:2020-05-22 06h40.19 utc:4497: sql_open: PQconnectPoll failed
md manage:WARNING:2020-05-22 06h40.19 utc:4497: sql_open: PQerrorMessage (conn): FATAL:  could not open relation mapping file "global/pg_filenode.map": Too many open files in system
md manage:WARNING:2020-05-22 06h40.19 utc:4497: init_manage_process: sql_open failed
md manage:WARNING:2020-05-22 06h40.22 utc:4502: sql_open: PQconnectPoll failed
md manage:WARNING:2020-05-22 06h40.22 utc:4502: sql_open: PQerrorMessage (conn): FATAL:  could not open relation mapping file "global/pg_filenode.map": Too many open files in system
md manage:WARNING:2020-05-22 06h40.22 utc:4502: init_manage_process: sql_open failed
md manage:   INFO:2020-05-22 06h41.43 utc:403: Updating /usr/local/var/lib/gvm/scap-data/nvdcve-2.0-2005.xml
md manage:   INFO:2020-05-22 06h44.05 utc:403: Updating /usr/local/var/lib/gvm/scap-data/nvdcve-2.0-2002.xml
md manage:   INFO:2020-05-22 06h45.06 utc:403: Updating /usr/local/var/lib/gvm/scap-data/nvdcve-2.0-2004.xml
md manage:   INFO:2020-05-22 06h46.01 utc:403: Updating /usr/local/var/lib/gvm/scap-data/nvdcve-2.0-2006.xml
md manage:   INFO:2020-05-22 06h47.19 utc:403: Updating /usr/local/var/lib/gvm/scap-data/nvdcve-2.0-2010.xml
md manage:   INFO:2020-05-22 06h48.56 utc:403: Updating OVAL data
md manage:   INFO:2020-05-22 06h49.03 utc:403: Updating /usr/local/var/lib/gvm/scap-data/oval/5.10/org.mitre.oval/c/oval.xml
md manage:   INFO:2020-05-22 06h49.03 utc:403: Updating /usr/local/var/lib/gvm/scap-data/oval/5.10/org.mitre.oval/m/oval.xml
md manage:   INFO:2020-05-22 06h49.03 utc:403: Updating /usr/local/var/lib/gvm/scap-data/oval/5.10/org.mitre.oval/v/family/ios.xml
md manage:   INFO:2020-05-22 06h49.05 utc:403: Updating /usr/local/var/lib/gvm/scap-data/oval/5.10/org.mitre.oval/v/family/pixos.xml
md manage:   INFO:2020-05-22 06h49.05 utc:403: Updating /usr/local/var/lib/gvm/scap-data/oval/5.10/org.mitre.oval/p/oval.xml
==> /usr/local/var/log/gvm/gsad.log <==
gsad  gmp:MESSAGE:2020-05-22 06h53.08 UTC:434: Authentication success for 'admin' from 172.17.0.1
==> /usr/local/var/log/gvm/gvmd.log <==
md manage:   INFO:2020-05-22 06h55.22 utc:403: Updating /usr/local/var/lib/gvm/scap-data/oval/5.10/org.mitre.oval/i/oval.xml
md manage:   INFO:2020-05-22 06h55.25 utc:403: Updating /usr/local/var/lib/gvm/scap-data/oval/5.10/org.mitre.oval/v/family/macos.xml
md manage:   INFO:2020-05-22 06h55.26 utc:403: Updating /usr/local/var/lib/gvm/scap-data/oval/5.10/org.mitre.oval/v/family/unix.xml
==> /usr/local/var/log/gvm/openvas.log <==
sd   main:MESSAGE:2020-05-22 06h56.01 utc:4065: Finished testing XX.XXX.X.XX. Time : 1124.59 secs
md manage:   INFO:2020-05-22 06h56.16 utc:403: Updating /usr/local/var/lib/gvm/scap-data/oval/5.10/org.mitre.oval/v/family/windows.xml
md manage:   INFO:2020-05-22 06h57.37 utc:403: Updating user OVAL definitions.
md manage:   INFO:2020-05-22 06h57.37 utc:403: Updating CVSS scores and CVE counts for CPEs
md manage:   INFO:2020-05-22 07h07.02 utc:403: Updating CVSS scores for OVAL definitions
md manage:   INFO:2020-05-22 07h07.12 utc:403: Updating placeholder CPEs
md manage:   INFO:2020-05-22 07h11.07 utc:403: update_scap: Updating SCAP info succeeded
md manage:WARNING:2020-05-22 07h11.17 utc:403: sql_exec_internal: PQexec failed: ERROR:  relation "cert_bund_advs" does not exist
LINE 1: SELECT EXISTS (SELECT * FROM cert_bund_advs  WHERE creation_...
                                     ^
 (7)
md manage:WARNING:2020-05-22 07h11.17 utc:403: sql_exec_internal: SQL: SELECT EXISTS (SELECT * FROM cert_bund_advs  WHERE creation_time        > coalesce (CAST ((SELECT value FROM meta                           WHERE name                                 = 'cert_check_time')                          AS INTEGER),                    0));
md manage:WARNING:2020-05-22 07h11.17 utc:403: sql_x_internal: sql_exec_internal failed
gsad  gmp:MESSAGE:2020-05-22 07h14.59 UTC:434: Authentication success for 'admin' from 172.17.0.1
sd   main:MESSAGE:2020-05-22 07h25.20 utc:4066: Finished testing XX.XXX.X.XX. Time : 2883.88 secs
sd   main:MESSAGE:2020-05-22 07h25.48 utc:4064: Finished testing XX.XXX.X.XX. Time : 2911.87 secs
sd   main:MESSAGE:2020-05-22 07h25.48 utc:4007: Test complete
sd   main:MESSAGE:2020-05-22 07h25.48 utc:4007: Total time to scan all hosts : 2921 seconds
==> /usr/local/var/log/gvm/ospd-openvas.log <==
2020-05-22 07:25:57,415 OSPD - openvas: INFO: (ospd.ospd) XX.XXX.X.XX, XX.XXX.X.XX, XX.XXX.X.XX: Host scan finished.
2020-05-22 07:25:58,532 OSPD - openvas: INFO: (ospd.ospd) 98661285-ef61-4a73-b894-731d2b1a43ab: Scan finished.
event task:MESSAGE:2020-05-22 07h27.01 UTC:3016: Status of task Konveodev Scan (626541f7-a2cd-40f7-bb1c-39bca9e67132) has changed to Done

Once the scan completed, when trying to access the scan result there is no information displayed (refer below screenshot). Is this caused by the error above?
<img width="1191" alt="1" src="https://user-images.githubusercontent.com/40619793/82645180-0e035f80-9c45-11ea-817f-1bac139356cd.png">

<img width="1180" alt="2" src="https://user-images.githubusercontent.com/40619793/82645200-15c30400-9c45-11ea-92fb-b2a57767856c.png">

We also noticed the load time for SecInfo data are significantly longer than before. Is this expected too?

**Comment by pixelsquared** [comment]
It looks like the database is running out of file descriptors. Try to increase the limit on max file descriptors on your host system.

For the results not showing, click the black X to the right of the filter text box.

**Comment by syafsss** [comment]
Hi, 

I have downloaded the latest [commit](https://github.com/Secure-Compliance-Solutions-LLC/GVM-Docker/commit/fc59bf363e4c901fe7dc96618635879dfe6d01f3) and perform another round of scan with it. I noticed I got some warning regarding DB once the GSA is ready for use:
> ++++++++++++++++++++++++++++++++++++++++++++++
> +Your GVM 11 container is now ready to use! +
> ++++++++++++++++++++++++++++++++++++++++++++++
> ++++++++++++++++
> +Tailing logs +
> ++++++++++++++++
> ==> /usr/local/var/log/gvm/gsad.log <==
> gsad main:MESSAGE:2020-06-03 06h02.13 utc:421: Starting GSAD version 9.0.1
> ==> /usr/local/var/log/gvm/gvmd.log <==
> md manage: INFO:2020-06-03 06h02.03 utc:400: update_scap: Initializing SCAP database
> md manage: INFO:2020-06-03 06h02.03 utc:402: sync_cert: Updating data from feed
> md manage: INFO:2020-06-03 06h02.03 utc:402: update_dfn_xml: dfn-cert-2019.xml
> md manage: INFO:2020-06-03 06h02.03 utc:402: Updating /usr/local/var/lib/gvm/cert-data/dfn-cert-2019.xml
> md manage: INFO:2020-06-03 06h02.03 utc:400: update_scap: Updating data from feed
> md manage: INFO:2020-06-03 06h02.03 utc:400: Updating CPEs
> md main:MESSAGE:2020-06-03 06h02.08 utc:412: Greenbone Vulnerability Manager version 9.0.1 (DB revision 221)
> md manage: INFO:2020-06-03 06h02.08 utc:412: Creating user.
> md manage:WARNING:2020-06-03 06h02.08 utc:412: database must be initialised from scanner
> md manage: INFO:2020-06-03 06h02.13 utc:401: OSP service has newer VT status (version 202006020940) than in database (version (null), 0 VTs). Starting update ...

Then I proceed to wait for all SecInfo data to finish loading, but there is error related to CERT-Bund (no target, scan config or scan task created yet at this point): 
> md manage: INFO:2020-06-03 07h01.40 utc:400: Updating user OVAL definitions.
> md manage: INFO:2020-06-03 07h01.40 utc:400: Updating CVSS scores and CVE counts for CPEs
> md manage: INFO:2020-06-03 07h11.08 utc:400: Updating CVSS scores for OVAL definitions
> md manage: INFO:2020-06-03 07h11.22 utc:400: Updating placeholder CPEs
> md manage: INFO:2020-06-03 07h15.19 utc:400: update_scap: Updating SCAP info succeeded
> md manage:WARNING:2020-06-03 07h15.28 utc:400: sql_exec_internal: PQexec failed: ERROR: relation "cert_bund_advs" does not exist
> LINE 1: SELECT EXISTS (SELECT * FROM cert_bund_advs WHERE creation_...
> ^
> (7)
> md manage:WARNING:2020-06-03 07h15.28 utc:400: sql_exec_internal: SQL: SELECT EXISTS (SELECT * FROM cert_bund_advs WHERE creation_time > coalesce (CAST ((SELECT value FROM meta WHERE name = 'cert_check_time') AS INTEGER), 0));
> md manage:WARNING:2020-06-03 07h15.28 utc:400: sql_x_internal: sql_exec_internal failed
> ==> /usr/local/var/log/gvm/gsad.log <==

Then when I finally run a scan, I still got the same error (as the first post above). My Docker also subsequently froze, so I have to restart it. Im not sure if this error is related to the DB warnings mentioned above or my Docker version (The same target successfully scanned using MacOS High Sierra v10.13.6 and Docker Desktop v2.3.0.2 (45183) without below error displayed): 
> ==> /usr/local/var/log/gvm/openvas.log <==
> sd   main:MESSAGE:2020-06-03 04h16.45 utc:3365: openvas 7.0.1 started
> sd   main:MESSAGE:2020-06-03 04h16.57 utc:3365: Starts a new scan. Target(s) : XX.XX.XX.XXX, XX.XX.XXX.XXX, XX.XXX.X.XX, with max_hosts = 20 and max_checks = 4
> sd   main:MESSAGE:2020-06-03 04h16.57 utc:3429: Testing XX.XX.XXX.XXX (Vhosts: ec2-XX.XX.XX.XXX-amazonaws.com) [3429]
> sd   main:MESSAGE:2020-06-03 04h16.57 utc:3427: Testing XX.XX.XXX.XXX (Vhosts: ec2-XX.XX.XX.XXX-amazonaws.com) [3427]
> sd   main:MESSAGE:2020-06-03 04h16.57 utc:3428: Testing XX.XX.XXX.XXX (Vhosts: ec2-XX.XX.XX.XXX-amazonaws.com) [3428]
> ==> /usr/local/var/log/gvm/gvmd.log <==
> md manage:WARNING:2020-06-03 04h17.06 utc:3494: sql_open: PQconnectPoll failed
> md manage:WARNING:2020-06-03 04h17.06 utc:3494: sql_open: PQerrorMessage (conn): FATAL:  could not open relation mapping file "global/pg_filenode.map": Too many open files in system
> md manage:WARNING:2020-06-03 04h17.49 utc:3567: init_manage_process: sql_open failed
> md manage:   INFO:2020-06-03 04h17.50 utc:386: Updating /usr/local/var/lib/gvm/scap-data/nvdcve-2.0-2008.xml
> md manage:WARNING:2020-06-03 04h17.52 utc:3571: sql_open: PQconnectPoll failed

**For additional information, I am using:** 
- OS: macOS Catalina v10.15.5
- Docker: 
-- Docker Desktop v2.3.0.3 (45519) Stable
-- Docker Engine v19.03.8
-- Compose v1.25.5
-- Notary v0.6.1

**Another additional info:**
1. The RSYNC service `feed.openvas.org` will be shut down **September 30th 2020**. Ref [here](https://community.greenbone.net/t/community-feed-url-consolidation/5338) based on this [post](https://community.greenbone.net/t/new-releases-for-gvm-10-and-gvm-11-2020-05-13/5355). 
2. Python gvm v1.5.0 is now available. Ref [here](https://github.com/greenbone/python-gvm/releases/tag/v1.5.0).

**Comment by pixelsquared** [comment]
Thanks for the heads up about the feed being shut down. I have opened a new issue for that: #32 


For the database error it looks like you need to increase the max open file limit. You may need to disable system integrity protection (SIP) for the steps below but remember to reactivate SIP after you have completed the steps below.

To disable SIP:
You need to restart your Mac, and hold Command-R until the Apple logo appears. When Recovery Mode has loaded, open Terminal and run this command:
```
csrutil disable
```

1. In /Library/LaunchDaemons create a file named limit.maxfiles.plist and paste the following in:
```
<?xml version="1.0" encoding="UTF-8"?>  
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN"  
        "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">  
  <dict>
    <key>Label</key>
    <string>limit.maxfiles</string>
    <key>ProgramArguments</key>
    <array>
      <string>launchctl</string>
      <string>limit</string>
      <string>maxfiles</string>
      <string>65536</string>
      <string>524288</string>
    </array>
    <key>RunAtLoad</key>
    <true/>
    <key>ServiceIPC</key>
    <false/>
  </dict>
</plist>
```

2. Change the owner of your new file:
```
sudo chown root:wheel /Library/LaunchDaemons/limit.maxfiles.plist
```
3. Load these new settings:
```
sudo launchctl load -w /Library/LaunchDaemons/limit.maxfiles.plist
```
4. Finally, check that the limits are correct:
```
launchctl limit maxfiles
```

To enable SIP:
You need to restart your Mac, and hold Command-R until the Apple logo appears. When Recovery Mode has loaded, open Terminal and run this command:
```
csrutil enable
```

**Comment by syafsss** [comment]
Hi Joshua, thank you for your detailed suggestion. When I tried to perform scan using the docker image with latest [commit](https://github.com/Secure-Compliance-Solutions-LLC/GVM-Docker/commit/c39410ec7eb9f6e4e95c33f8a2b9e55337fed555), the database error is no longer reproducible. I did not do any changes in my machine yet, it might have something to do with the updated NVT url. 

The `md manage:WARNING:2020-06-03 06h02.08 utc:412: database must be initialised from scanner` warning and `cert_bund_advs` error however is still reproducible. 
Not sure if this is an existing issue from Greenbone, and since it didnt affect my scanning I will close this thread. 

Thanks so much for your help.
